jerry-core/api/jerry.c

```diff
@@ -254,6 +255,34 @@ jerry_gc (void)
   ecma_gc_run (JMEM_FREE_UNUSED_MEMORY_SEVERITY_LOW);
 } /* jerry_gc */
 
+#ifdef JMEM_STATS
+void jmem_heap_get_stats (jmem_heap_stats_t *);
+#endif
+/**
+ * Get heap memory stats.
+ *
+ * @return true - get the heap stats successful
+ *         false - otherwise. Usually it is because the MEM_STATS feature is not enabled.
+ */
+bool
+jerry_get_memory_stats (jerry_heap_stats_t *out_stats_p) /**< [out] heap memory stats */
+{
+#ifdef JMEM_STATS
+  jmem_heap_stats_t jmem_heap_stats = {0};
+  jmem_heap_get_stats (&jmem_heap_stats);
+
+  out_stats_p->version = 1;
```

Comment:
I would make sure to completely initialize the struct.
Personally I like doing this as such:

```
*out_stats_p = (jerry_heap_stats_t) {
  .size-> = ...,
 // ...
 // if more fields are added to the struct in the future, but we forget them here,
 // they will at least be initialized to zero (instead of being garbage).
};
```